pkg/apis/batch/v2alpha1/register.go

```diff
@@ -17,34 +17,41 @@ limitations under the License.
 package v2alpha1
 
 import (
-	"k8s.io/kubernetes/pkg/api/unversioned"
 	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/apimachinery/announced"
 	"k8s.io/kubernetes/pkg/runtime"
 	versionedwatch "k8s.io/kubernetes/pkg/watch/versioned"
 )
 
-// GroupName is the group name use in this package
-const GroupName = "batch"
-
-// SchemeGroupVersion is group version used to register these objects
-var SchemeGroupVersion = unversioned.GroupVersion{Group: GroupName, Version: "v2alpha1"}
+var (
+	// VersionArgs is the canonical place for facts about this group version.
+	VersionArgs = &announced.GroupVersionFactoryArgs{
+		GroupName:   "batch",
+		VersionName: "v2alpha1",
+		VersionedObjects: []runtime.Object{
+			&Job{},
+			&JobList{},
+			&JobTemplate{},
+			&ScheduledJob{},
+			&ScheduledJobList{},
+			&v1.ListOptions{},
+			&v1.DeleteOptions{},
+		},
+		SchemeAdjustment: runtime.NewSchemeBuilder(
+			addDefaultingFuncs, addConversionFuncs,
+		).VersionedRegister(versionedwatch.AddToGroupVersion),
+	}
+	SchemeBuilder = VersionArgs.SchemeAdjustment
+)
 
 var (
-	SchemeBuilder = runtime.NewSchemeBuilder(addKnownTypes, addDefaultingFuncs, addConversionFuncs)
-	AddToScheme   = SchemeBuilder.AddToScheme
+	AddToScheme        = SchemeBuilder.AddToScheme
+	GroupName          = VersionArgs.GroupName
+	SchemeGroupVersion = VersionArgs.SchemeGroupVersion()
 )
 
-// Adds the list of known types to api.Scheme.
-func addKnownTypes(scheme *runtime.Scheme) error {
-	scheme.AddKnownTypes(SchemeGroupVersion,
-		&Job{},
-		&JobList{},
-		&JobTemplate{},
-		&ScheduledJob{},
-		&ScheduledJobList{},
-		&v1.ListOptions{},
-		&v1.DeleteOptions{},
-	)
-	versionedwatch.AddToGroupVersion(scheme, SchemeGroupVersion)
-	return nil
+func init() {
```

Comment:
That's coming in the future. The goal here was to make 0 effective change so I don't have to worry about breaking things. Then the PR that actually removes the init statements will be small and easy to reason about. Hopefully!